packages/iov-bns/src/bnsconnection.ts

```diff
@@ -745,6 +754,43 @@ export class BnsConnection implements AtomicSwapConnection {
     return votes;
   }
 
+  public async getDeposits(query: BnsDepositQuery): Promise<readonly BnsTermDepositNft[]> {
+    let keyPrefix = "deposit:";
+    let results: readonly Result[];
+    if (isBnsDepositsByDepositorQuery(query)) {
+      keyPrefix = "";
+      const rawAddress = decodeBnsAddress(query.depositor).data;
+      results = (await this.query("/deposits/depositor", rawAddress)).results;
+    } else if (isBnsDepositsByContractIdQuery(query)) {
+      keyPrefix = "";
+      results = (await this.query("/deposits/contract", query.depositContractId)).results;
+    } else if (isBnsDepositByDepositIdQuery(query)) {
+      results = (await this.query("/deposits", query.depositId)).results;
+    } else {
+      throw new Error("Unsupported query");
+    }
+
+    const parser = createParser(codecImpl.termdeposit.Deposit, keyPrefix);
```

Comment:
For unknown reason keyPrefix for `depositor` and `contract` indexes should be empty, because BNS returns result and this results key contains record id only and did not contain any prefix. Is this a bug or feature? @davepuchyr

Reply:
`/deposits/depositor` and `/deposits/contract` are both paths that point directly to indexes and, therefore, don't need a prefix.  `/deposits` points to a bucket whose keys are prefixed with `deposit:`, hence the need for a non-empty `keyPrefix` in the parser for the bucket query.  Does that make sense?

Reply:
But in [this case](https://github.com/iov-one/iov-core/blob/b271c195f199f59d96daa7c9ef4c935284c60337/packages/iov-bns/src/bnsconnection.ts#L777) index returns keys with prefix in it.

Reply:
Ah.  Piotr refactored the indexing algorithm.  Since `usernames` are deprecated I guess he (correctly) didn't seek backward compatibility.